packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -224,18 +262,29 @@ const DiagramEditor: React.FunctionComponent<{
             };
           }
         ),
-        measured: {
-          width: 100,
-          height: 200,
-        },
       })
     );
   }, [model?.collections]);
 
+  useEffect(() => {
+    if (nodes.length === 0) return;
+    const isInitialState = nodes.every(
+      (node) => isNaN(node.position.x) || isNaN(node.position.y)
+    );
```

Comment:
Does it need to be `every` check? Is there a case where we would consider a single node not having coordinates as a valid state that doesn't need position calculated?